WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -100,7 +101,10 @@ extension PointOfSaleAggregateModel {
 
     @MainActor
     func reload() async {
-        await itemsController.reload()
+        reloadTask?.cancel()
+        reloadTask = Task { @MainActor in
+            await itemsController.reload()
+        }
```

Comment:
Hmmm I think I can see what's going on now. The error we're getting is `explicitlyCancelled` – it looks like we're doing that by causing the view to redraw within the reload function. We call `itemListStateSubject.send(.loading(allItems))` when it's called, which redraws the view to show the loading state at the top... but doing that redraw means that SwiftUI cancels the refresh control.

Reply:
Yes the `explicitlyCancelled` cancelation that results in error state is the issue. Thanks for the better solution by not triggering redraw in https://github.com/woocommerce/woocommerce-ios/pull/14728/commits/1fb8df061cb1965ef071aa008aa63c49a17c26c8! From my testing with stores with 2 POS products in your PR https://github.com/woocommerce/woocommerce-ios/pull/14727, I occasionally still saw this error but only briefly because the pre-existing issue https://github.com/woocommerce/woocommerce-ios/issues/14735 where `loadNextItems` is also triggered when pulling to refresh a short list (this request isn't canceled and the items from this request update the items view state). Would be great to use `Observation` to fix this issue properly.